### ISSUE TITLE
pgpainless-cli decrypt --verify-with changes the return code based on whether a valid signature was found or not

### ISSUE THREAD

**Issue by dkg** [vanitasvitae/pgpainless#415]
[the SOP documentation](https://www.ietf.org/archive/id/draft-dkg-openpgp-stateless-cli-08.html#name-decrypt-decrypt-a-message) says:

> The return code of sop decrypt is not affected by the results of signature verification. The caller MUST check the returned `VERIFICATIONS` to confirm signature status. An empty `VERIFICATIONS` output indicates that no valid signatures were found.

But `pgpainless-cli` appears to change the return code based on whether a signature was present or not:

```
0 dkg@alice:/tmp/cdtemp.4jVsy5$ pgpainless-cli generate-key foo > foo.key
0 dkg@alice:/tmp/cdtemp.4jVsy5$ pgpainless-cli extract-cert < foo.key > foo.cert
0 dkg@alice:/tmp/cdtemp.4jVsy5$ echo test | pgpainless-cli encrypt foo.cert > test.msg
0 dkg@alice:/tmp/cdtemp.4jVsy5$ pgpainless-cli decrypt foo.key  < test.msg 
test
0 dkg@alice:/tmp/cdtemp.4jVsy5$ pgpainless-cli decrypt --verify-out test.verifs --verify-with foo.cert foo.key  < test.msg 
test
No verifiable signature found.
3 dkg@alice:/tmp/cdtemp.4jVsy5$ pgpainless-cli version --extended
PGPainless-SOP 1.3.16
https://codeberg.org/PGPainless/pgpainless/src/branch/master/pgpainless-sop

Implementation of the Stateless OpenPGP Protocol Version 04
https://datatracker.ietf.org/doc/html/draft-dkg-openpgp-stateless-cli-04

Based on pgpainless-core 1.3.16
https://pgpainless.org

Using Bouncy Castle 1.72
https://www.bouncycastle.org/java.html
0 dkg@alice:/tmp/cdtemp.4jVsy5$ 
```

**Comment by vanitasvitae** [comment]
Good findings!
For some reason I haven't considered the `VERIFICATIONS` output here, which is a way better way of signalling than the exit code!

**Comment by vanitasvitae** [comment]
Sidenote: This is actually a `sop-java` issue.
I fixed it in `sop-java 4.1.2`, so we can have a patch in Debian and will port the fix to newer releases as well :)

**Comment by vanitasvitae** [comment]
I released fixes for `4.1.2`, `5.0.2`, `6.0.1`, `7.0.1` and `8.0.1`.
I guess I will apply these fixes to `pgpainless-cli` versions `1.3.18` for Debian, and the latest main, which will eventually become `2.0`.

**Comment by vanitasvitae** [comment]
I just released PGPainless version `1.3.18`, `1.4.6`, `1.5.7` and `1.6.3` with a fix :)
